pkg/blueprint/customizations.go

```diff
@@ -27,6 +27,11 @@ type Customizations struct {
 	FIPS               *bool                          `json:"fips,omitempty" toml:"fips,omitempty"`
 	ContainersStorage  *ContainerStorageCustomization `json:"containers-storage,omitempty" toml:"containers-storage,omitempty"`
 	Installer          *InstallerCustomization        `json:"installer,omitempty" toml:"installer,omitempty"`
+	ImportRPMGPGKey    *ImportRPMGPGKeyCustomization  `json:"import_rpm_gpg_key,omitempty" toml:"import-rpm-gpg-key,omitempty"`
```

Comment:
Not the best name, I'm open to suggestions :innocent:

Reply:
I'm more concerned about the TOML / JSON property name. The struct is hopefully fine. Nevertheless, I appreciate the suggestion.

Reply:
> I'm more concerned about the TOML / JSON property name

Is there a reason to have one property in snake case and the other in kebab case?

Slightly further up we have both in snake case for `installation_device`:
https://github.com/osbuild/images/blob/eff7b17b216ee87b85a52ce3820a6829c92873d9/pkg/blueprint/customizations.go#L20

Reply:
I'd recommend putting it one level down into a category.  I don't have a great idea for this, but I reflexively don't like adding top-level customizations for something this specific.

Now, that said, I don't have a good idea what the category should be.  I wonder if it makes sense to have an `rpm` category.  Is it possible we might need rpm configs in the future?
So given all that, I'm thinking maybe the customization could be, in json:
```json
{
  "rpm": {
    "import_keys": {
      "keys": [
        "actual key strings"
      ],
      "files": [
        "paths to files in the tree"
      ]
    }
  }
}
```
which would be a bit more tidy, enables pasting keys straight into the image (which might be convenient), and leaves room for more rpm-related configurations.

It might have been nice to have a package-related customization block which would also include the Repository customizations.

Reply:
A more extreme version:
```json
{
  "rpm": {
    "gpg": {
      "import": {
        "keys": [],
        "paths": []
      }
    }
  }
}
```

Reply:
> I'd recommend putting it one level down into a category. I don't have a great idea for this, but I reflexively don't like adding top-level customizations for something this specific.
> 
> Now, that said, I don't have a good idea what the category should be. I wonder if it makes sense to have an `rpm` category. Is it possible we might need rpm configs in the future? So given all that, I'm thinking maybe the customization could be, in json:
> 
> ```json
> {
>   "rpm": {
>     "import_keys": {
>       "keys": [
>         "actual key strings"
>       ],
>       "files": [
>         "paths to files in the tree"
>       ]
>     }
>   }
> }
> ```
> 
> which would be a bit more tidy, enables pasting keys straight into the image (which might be convenient), and leaves room for more rpm-related configurations.

Makes sense...

I've already defined the struct in a way, so that pasting literal key strings would be possible (there is `from_tree` and there could be something else for strings). rpm-related config is not something I foresee us adding, but it won't hurt to keep the options open :wink: 

> It might have been nice to have a package-related customization block which would also include the Repository customizations, but that ship has sailed.

Agreed, I had a similar idea, but adding it to package customization, which is not really a customization but a top level property of a BP... :confused:

Reply:
I went with the first "less extreme" version :sweat_smile:

Reply:
> It might have been nice to have a package-related customization block which would also include the Repository customizations, but that ship has sailed.

Definitely would have been nice, oh well 😅

Reply:
> I went with the first "less extreme" version 😅

Makes sense :)